src/Installer/DownloadInterceptor.php

```diff
@@ -32,14 +33,12 @@ public function __construct(DownloaderInterface $downloader, Filesystem $filesys
     }
 
     /**
-     * @param PackageInterface $package
-     * @param string           $path
+     * @param PhivePackageInterface $package
      */
-    public function download(PackageInterface $package, $path) {
+    public function download(PhivePackageInterface $package) {
         $tmpFile = $this->downloader->download($package, sys_get_temp_dir());
-        $this->filesystem->rename($tmpFile, $path . '/' . basename($package->getDistUrl()));
+        $this->filesystem->rename($tmpFile, $package->getInstallationPath());
 
-        // be sure that phive.phar is executable
-        Silencer::call('chmod', $path . '/' . basename($package->getDistUrl()), 0777 & ~umask());
+        Silencer::call('chmod', $package->getInstallationPath(), 0777 & ~umask());
```

Comment:
I can. But thats [composer's way to do this](https://github.com/composer/composer/search?utf8=%E2%9C%93&q=Silencer&type=). :)
Would you still change it @theseer?